espflash/src/targets/flash_target/ram.rs

```diff
@@ -36,6 +36,15 @@ impl FlashTarget for RamTarget {
         Ok(())
     }
 
+    fn write_segment_sdm(
+        &mut self,
+        _connection: &mut Connection,
+        _segment: Segment<'_>,
+        _progress: &mut Option<&mut dyn ProgressCallbacks>,
+    ) -> Result<(), Error> {
+        todo!()
```

Comment:
If we can't implement this, this should return a nice error, not panic.

Reply:
If it can be implemented at some point, please also add a TODO with a relevant issue.